### PR TITLE
Mod compositor consolidate shell surfaces

### PR DESCRIPTION
**Pull request by neithernut**
Targets #538.

### CODE REVIEW

File: src/compositor/wayland/xdg_surface.c [hunk 1]
```diff
@@ -357,7 +357,9 @@ ws_wayland_xdg_surface_new(
         return NULL;
     }
 
-    int retval = ws_wayland_obj_init(&self->shell.wl_obj, resource);
+    int retval;
+    retval = ws_abstract_shell_surface_init(&self->shell, resource,surface,
```
Comment (matthiasbeyer) [inline src/compositor/wayland/xdg_surface.c]: Missing whitespace after `,` here.

**Comment by matthiasbeyer** [comment]
Besides the one whitespace annotation: ACK.

**Comment by neithernut** [comment]
I'll also introduce some additional changes.

**Comment by neithernut** [comment]
I'm done here for now.

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**